extension/headerssetterextension/README.md

```diff
@@ -17,27 +17,40 @@ header to the value extracted from the context.
 
 The following settings are required:
 
-- `headers`: a list of header configuration objects that specify headers and 
-   their value sources. Each configuration object has the following properties:
-    - `key`: the header name
-    - `value`: the header value is looked up from the `value` property of the
-       extension configuration
-    - `from_context`: the header value is looked up from the request metadata,
-       such as HTTP headers, using the property value as the key (likely a header name)
+- `headers`: a list of header configuration objects that specify headers and
+  their value sources. Each configuration object has the following properties:
+    - `key`: The header name.
+    - `action`: An action to perform on the header. Supported actions are:
+        - `insert`: Inserts the new header if it does not exist.
```

Comment:
If you decide to make this the default action, mark it as such here.

Reply:
I'd be inclined to always require an action actually.